### PR TITLE
using service workers to render page thumbnails

### PR DESCRIPTION
**Pull request by marcoscaceres**

### CODE REVIEW

**Comment by marcoscaceres** [comment]
@sarracini, can you give this a once over, make sure that everything got brought over from #10?

**Comment by sarracini** [comment]
r+, looks good

**Comment by sarracini** [comment]
r+, looks good

**Comment by marcoscaceres** [comment]
Argh :( This branch is different from `pagethumbs_message_passing`. 

That doesn't seem right :(

File: browser/components/newtab/RemoteAboutNewTab.jsm [hunk 1]
```diff
@@ -316,14 +333,29 @@ let RemoteAboutNewTab = {
...
-    });
+
+    img.onload = function (e) {
+      canvas.width = img.naturalWidth;
+      canvas.height = img.naturalHeight;
+      var ctx = canvas.getContext("2d");
+      ctx.drawImage(this, 0, 0, this.naturalWidth, this.naturalHeight);
+      canvas.toBlob(function (blob) {
+        let host = new URL(message.data.link.url).host;
+        AboutNewTab.pageListener.sendAsyncMessage("NewTab:RegularThumbnailURI", {
```
Comment (oyiptong) [inline browser/components/newtab/RemoteAboutNewTab.jsm]: Should be `RemoteAboutNewTab.pageListener`

**Pull request merged**